test/Sema/availability_implicitly_unnecessary.swift

```diff
@@ -0,0 +1,12 @@
+// Ensure that the `unnecessary check` availability warning is emitted when unnecessary due to
+// scope's explicit annotation
```

Comment:
Should this be the description for the other test?

Reply:
Yes, thanks for catching this!